salt/modules/virt.py

```diff
@@ -2306,11 +2346,10 @@ def update(
 
         # Existing tag is found and values don't match
         if found_tag is not None and found_tag.text != boot_tag_value:
-
             # If the existing tag is found, but the new value is None
             # remove it. If the existing tag is found, and the new value
             # doesn't match update it. In either case, mark for update.
-            if boot_tag_value is None and boot is not None and parent_tag is not None:
+            if boot_tag_value == "None" and boot is not None and parent_tag is not None:
```

Comment:
strange... why `"None"` as a string? that shouldn't happen

Reply:
It seems like passing `None` in the sls file is handled as a string.

Reply:
@gqlo can you provide some insight into a situation where you would be passing `None` as an argument in an SLS file?

Reply:
@garethgreenaway  Hi garethgreen I added the feature to support booting VM with `uefi` in this PR #56613, in the case where we want to change back to BIOS boot or remove booting directly from a specified kernel, passing `None` as an argument would work. Do you have better ideas on this ?

Reply:
The SLS / Yaml mapping for python's `None` is [Null](https://yaml.org/type/null.html)